turtle_tf2_cpp/src/static_turtle_tf2_broadcaster.cpp

```diff
@@ -0,0 +1,102 @@
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <geometry_msgs/msg/transform_stamped.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2_ros/static_transform_broadcaster.h>
+
+#include <memory>
+
+using std::placeholders::_1;
+
+class StaticFramePublisher : public rclcpp::Node
+{
+public:
+  explicit StaticFramePublisher(char * transformation[])
+  : Node("static_turtle_tf2_broadcaster")
+  {
+    tf_publisher_ = std::make_shared<tf2_ros::StaticTransformBroadcaster>(this);
+
+    // Publish static transforms once at startup
+    this->make_transforms(transformation);
+  }
+
+private:
+  void make_transforms(char * transformation[])
+  {
+    rclcpp::Time now;
+    geometry_msgs::msg::TransformStamped t;
+
+    t.header.stamp = now;
+    t.header.frame_id = "world";
+    t.child_frame_id = transformation[1];
+
+    t.transform.translation.x = atof(transformation[2]);
+    t.transform.translation.y = atof(transformation[3]);
+    t.transform.translation.z = atof(transformation[4]);
+    tf2::Quaternion q;
+    q.setRPY(
+      atof(transformation[5]),
+      atof(transformation[6]),
+      atof(transformation[7]));
+    t.transform.rotation.x = q.x();
+    t.transform.rotation.y = q.y();
+    t.transform.rotation.z = q.z();
+    t.transform.rotation.w = q.w();
+
+    tf_publisher_->sendTransform(t);
+  }
+  std::shared_ptr<tf2_ros::StaticTransformBroadcaster> tf_publisher_;
+};
+
+int main(int argc, char * argv[])
+{
+  auto logger = rclcpp::get_logger("logger");
+
+  // Obtain parameters from command line arguments
+  if (argc != 8) {
+    RCLCPP_INFO(
+      logger, "Invalid number of parameters\nusage: "
+      "ros2 run learning_tf2_cpp static_turtle_tf2_broadcaster "
+      "child_frame_name x y z roll pitch yaw");
+    return 1;
+  }
+  if (strcmp(argv[1], "world") == 0) {
```

Comment:
It'd be nice to have a comment explaining that you use the frame id "world" elsewhere in the example, which is why you check that it isn't used here.